docs/MAINTAINERS.md

```diff
@@ -1,52 +1,138 @@
 
-- Add a new entry to `CHANGELOG.md` by running `make changelog`.
+### Flight Plan
 
-- Update `screenshot.png` so it displays the latest version in the bottom
-right corner.
+#### Preparation
+
+- [Prepare the new version](#preparing-a-new-version)
+- [Generate build artefacts](#generating-binaries) (binaries, archives, etc.)
+- [Draft a release on GitHub](https://github.com/resin-io/etcher/releases)
+    - Upload build artefacts to GitHub release draft
+
+#### Testing
+
+- Test the prepared release and build artefacts properly on **all supported operating systems** to prevent regressions that went uncaught by the CI tests
+- If regressions or other issues arise, create issues on the repository for each one, and decide whether to fix them in this release (meaning repeating the process up until this point), or to follow up with a patch release
 
+#### Publishing
+
+- Publish release draft on GitHub
+- Upload build artefacts to Bintray & Amazon S3
+- Post release note to forums
+- If this release packs noteworthy major changes,  
+maybe write a blog post about it as well
+
+### Preparing a New Version
+
+- Create & hop onto a new release branch, i.e. `release-1.0.0`
+- Bump the version number in the `package.json`'s `version` property.
+- Bump the version number in the `npm-shrinkwrap.json`'s `version` property
+- Add a new entry to `CHANGELOG.md` by running `make changelog`
+- Update `screenshot.png` so it displays the latest version in the bottom
+right corner
 - Revise the `updates.semverRange` version in `package.json`
+- Commit the changes with the version number as the commit title, including the `v` prefix, to `master`. For example:
 
-- Commit the changes with the version number as the commit title, including the
-`v` prefix, to `master`. For example:
+**NOTE:** The version **MUST** be prefixed with a "v"
 
-```sh
+```bash
 git commit -m "v1.0.0" # not 1.0.0
 ```
 
-- Create an annotated tag for the new version. The commit title should equal
-the annotated tag name. For example:
+- Create an annotated tag for the new version. The commit title should equal the annotated tag name. For example:
 
-```sh
+```bash
 git tag -a v1.0.0 -m "v1.0.0"
 ```
 
 - Push the commit and the annotated tag.
 
-```sh
+```bash
 git push
 git push --tags
 ```
 
-Upgrading Electron
-------------------
+- Open a pull request against `master` titled "Release v1.0.0"
+
+### Generating binaries
+
+**Environment**
+
+Make sure to set the analytics tokens when generating production release binaries:
+
+```bash
+export ANALYTICS_SENTRY_TOKEN="xxxxxx"
+export ANALYTICS_MIXPANEL_TOKEN="xxxxxx"
+```
+
+#### Linux
+
+##### Clean dist folder
+
+**NOTE:** Make sure to adjust the path as necessary (here the Etcher repository has been cloned to `/home/$USER/code/etcher`)
+
+```bash
+docker run -v "/home/$USER:/home/$USER" "busybox rm -rf /home/$USER/code/etcher/release /home/$USER/code/etcher/node_modules /home/$USER/code/etcher/build /home/$USER/code/etcher/dist"
```

Comment:
Or `make distclean` ;-)  (and there's no need to use docker for this step either)

Reply:
There is – when doing the builds on misc1, everything under `dist` will be owned by root, and `make distclean` will not work.

Reply:
Fair enough. You could still replace this with `./scripts/build/docker/run-command.sh -r x64 -s . -c "make distclean"` though.

Reply:
Yeah, I think @lurch's last comment should work.